edb/server/dbview/dbview.pyx

```diff
@@ -191,26 +194,35 @@ cdef class Database:
                 await self.tenant.evict_query_cache(self.name, keys)
 
             # Now, populate the cache
-            query_req, units = await self._cache_queue.get()
-            # TODO: Should we do any sort of error handling here?
-            if (
-                len(units) == 1
+            # Empty the queue, for batching reasons.
+            ops = [await self._cache_queue.get()]
+            while not self._cache_queue.empty():
+                ops.append(self._cache_queue.get_nowait())
```

Comment:
I'd add a comment that this is safe for asyncio queues (unlike thread queues, where the empty check /get_nowait combo is not guaranteed to succeed due to races)

Reply:
(I froze here for a few moments thinking, mistakingly, that your code is incorrect)

Reply:
Oh yeah; I felt naughty while doing this, but should have added a comment